File: UI_testing/test_node_selenium/test_sonar.js
const webDriver = require('selenium-webdriver')
const SonarqubePage = require('./page_objects/sonarqube_page')
const constants = require('./constants')

jasmine.DEFAULT_TIMEOUT_INTERVAL = 50000

describe('sonarqube test', () => {
    let driver
    let sonarqubeDriver
    it('screenshot project', async () => {
        driver = new webDriver.Builder()
        .forBrowser('chrome')
        .build()
        try{
            sonarqubeDriver = new SonarqubePage(driver)
            await sonarqubeDriver.goto(constants.BASEURL)
    
            await sonarqubeDriver.clickProjectsTab()
            await sonarqubeDriver.clickProject(constants.PROJECT_NAME)
            await sonarqubeDriver.screenshotProject()
        }
        finally {
            await driver.quit()
        }
    })
})

File: UI_testing/test_python_selenium/test_sonar.py
from page_objects.sonarqube_page import SonarqubePage
import constants
import pytest

def test_sonar(chrome_web_driver):
    browser = chrome_web_driver
    sonarqube = SonarqubePage(browser)

    sonarqube.goto(constants.BASEURL)

    sonarqube.click_portfolio_tab()
    sonarqube.click_project(constants.PROJECT_NAME)
    sonarqube.screenshot_project('{}.png'.format(constants.PROJECT_NAME))

    browser.quit()
